tvm/src/pass/generate_reuse_buffer.cc

```diff
@@ -55,6 +55,99 @@ class ModulusRemover final : public IRMutator {
   std::map<const Variable*, Expr>& range_;
 };
 
+class CastRemover final : public IRMutator {
+ public:
+  CastRemover() {}
+
+  Expr Mutate_(const Cast* op, const Expr& e) {
+    return op->value;
+  }
+
+  Expr Mutate_(const Add* op, const Expr& e) {
+    Expr a = this->Mutate(op->a);
+    Expr b = this->Mutate(op->b);
+    if (const Cast* ca = a.as<Cast>()) {
+      a = ca->value;
+    }
+    if (const Cast* cb = b.as<Cast>()) {
+      b = cb->value;
+    }
+    return Add::make(a, b);
+  }
```

Comment:
1. Is it safe to remove the cast operations?
2. Seems the `CastRemover` does not enumerate all the operations but only several binary ones.

Reply:
I added some explanation below of why we do this, please see below the thread